packages/generator/src/engines/base.ts

```diff
@@ -133,3 +133,29 @@ export const getBaseEngine = (conf: ConfigResultWithHooks) => {
     layerNames,
   }
 }
+
+export interface PandaBaseEngine extends ConfigResultWithHooks {
+  isTemplateLiteralSyntax: boolean
+  studio: RequiredBy<NonNullable<ConfigResultWithHooks['config']['studio']>, 'outdir'>
+  hash: {
+    tokens: boolean | undefined
+    className: boolean | undefined
+  }
+  prefix: {
+    tokens: string | undefined
+    className: string | undefined
+  }
```

Comment:
Let's make the types package the source of truth for these types.

I'd imagine HashOptions, PrefixOptions type